src/Spice86/UserControls/AddressColumn.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace Spice86.UserControls;
+
+using Avalonia;
+using Avalonia.Media.TextFormatting;
+
+using AvaloniaHex.Rendering;
+
+using Spice86.Shared.Utils;
+
+using System;
+
+internal class AddressColumn : Column {
+    private Size _minimumSize;
+
+    static AddressColumn() {
+        IsUppercaseProperty.Changed.AddClassHandler<HexColumn, bool>(OnIsUpperCaseChanged);
+    }
+
+    /// <inheritdoc />
+    public override Size MinimumSize => _minimumSize;
+
+    /// <summary>
+    /// Defines the <see cref="IsUppercase"/> property.
+    /// </summary>
+    public static readonly StyledProperty<bool> IsUppercaseProperty =
+        AvaloniaProperty.Register<HexColumn, bool>(nameof(IsUppercase), true);
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the hexadecimal digits should be rendered in uppercase or not.
+    /// </summary>
+    public bool IsUppercase {
+        get => GetValue(IsUppercaseProperty);
+        set => SetValue(IsUppercaseProperty, value);
+    }
+
+    private static void OnIsUpperCaseChanged(HexColumn arg1, AvaloniaPropertyChangedEventArgs<bool> arg2) {
+        arg1.HexView?.InvalidateVisualLines();
+    }
+
+    /// <inheritdoc />
+    public override void Measure() {
+        if (HexView is null) {
+            _minimumSize = default;
+        } else {
+            TextLine dummy = CreateTextLine("00000000 (0xFFFF:0xFFFF):")!;
+            _minimumSize = new Size(dummy.Width, dummy.Height);
+        }
+    }
+
+    /// <inheritdoc />
+    public override TextLine? CreateTextLine(VisualBytesLine line) {
+        if (HexView is null)
+            throw new InvalidOperationException();
+
+        ulong address = line.Range.Start.ByteIndex;
+        ushort segment = ConvertUtils.ToAbsoluteSegment((uint)address);
```

Comment:
Is it useful to even display segment:offset if we can't use the same values that the game does?

Reply:
Not really, no. Remove it ?

Reply:
Removed. I use the OffsetColumn for the nuget package.